test_repo_version needs tune up?

I believe (didn't confirm anyhow, but daily testing just now started to fail / ref1) due to [10.20220724-5-g63cef2ae0](https://git.kitenet.net/index.cgi/git-annex.git/commit/?id=63cef2ae0b4069050fe43b14cd1656fbc843b9a4) from 25th:

```
______________________________ test_repo_version _______________________________
33840

33841
path1 = '/tmp/datalad_temp_test_repo_version9_18t4kn'
33842
path2 = '/tmp/datalad_temp_test_repo_versionkdcc3yeo'
33843
path3 = '/tmp/datalad_temp_test_repo_versionqd73k494'
33844

33845
    @with_tempfile
33846
    @with_tempfile
33847
    @with_tempfile
33848
    def test_repo_version(path1=None, path2=None, path3=None):
33849
        with swallow_logs(new_level=logging.INFO) as cm:
33850
            annex = AnnexRepo(path1, create=True, version=6)
33851
            assert_repo_status(path1, annex=True)
33852
            version = int(annex.config.get('annex.version'))
33853
            # Since git-annex 7.20181031, v6 repos upgrade to v7.
33854
            supported_versions = AnnexRepo.check_repository_versions()["supported"]
33855
            v6_lands_on = next(i for i in supported_versions if i >= 6)
33856
>           eq_(version, v6_lands_on)
33857

33858
/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/site-packages/datalad/support/tests/test_annexrepo.py:1301: 
33859
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
33860

33861
first = 10, second = 8, msg = None
33862

33863
    def assert_equal(first, second, msg=None):
33864
        if msg is None:
33865
>           assert first == second
33866
E           assert 10 == 8
```

so the logic of landing on the first supported is no longer true, and I think that is ok.  Most likely we can only generally test that we land on some supported version, not necessarily the first. WDYT?


<details>
<summary>ref1</summary> 

```shell
 327   T Jul 28 GitHub Actions  *-1.9* (5.5K/0) datalad/git-annex daily summary: 16 PASSED, 9 FAILED                                                                                 
 728 O T Jul 27 GitHub Actions  *-1.3* (4.7K/0) datalad/git-annex daily summary: 11 PASSED, 6 FAILED, 4 INCOMPLETE                                                                   
1260 O T Jul 26 GitHub Actions  *-1.9* (5.5K/0) datalad/git-annex daily summary: 16 PASSED, 9 FAILED                                                                                 
1658   T Jul 25 GitHub Actions  *-1.9* (5.1K/0) datalad/git-annex daily summary: 19 PASSED, 2 FAILED, 2 INCOMPLETE
```
</details>

--- a/datalad/support/annexrepo.py
+++ b/datalad/support/annexrepo.py
@@ -105,10 +105,6 @@ from .exceptions import (
 
 lgr = logging.getLogger('datalad.annex')
 
-# This is a map between an auto-upgradeable version and the version that it
-# upgrades to. It should track autoUpgradeableVersions in Annex.Version.
-_AUTO_UPGRADEABLE_VERSIONS = {v: 8 for v in range(3, 8)}
-
 
 class AnnexRepo(GitRepo, RepoInterface):
     """Representation of an git-annex repository.
@@ -1438,10 +1434,15 @@ class AnnexRepo(GitRepo, RepoInterface):
         if description is not None:
             opts += [description]
         if version is not None:
-            upgraded_version = _AUTO_UPGRADEABLE_VERSIONS.get(version)
-            if upgraded_version:
-                lgr.info("Annex repository version %s will be upgraded to %s",
-                         version, upgraded_version)
+            version = str(version)
+            supported_versions = AnnexRepo.check_repository_versions()['supported']
+            if version not in supported_versions:
+                first_supported_version = int(supported_versions[0])
+                if int(version) < first_supported_version:
+                    lgr.info("Annex repository version %s will be upgraded to %s or later version",
+                             version, first_supported_version)
+                    # and if it is higher than any supported -- we will just let git-annex to do
+                    # what it wants to do
             opts += ['--version', '{0}'.format(version)]
 
         # TODO: RM DIRECT?  or RF at least ?

--- a/datalad/support/tests/test_annexrepo.py
+++ b/datalad/support/tests/test_annexrepo.py
@@ -1300,20 +1300,27 @@ def test_repo_version_upgrade(path1=None, path2=None, path3=None):
         # but would switch to first upgradeable after
         Uversion = 6 if 6 in _GIT_ANNEX_VERSIONS_INFO["upgradable"] \
             else _GIT_ANNEX_VERSIONS_INFO["upgradeable"][0]
-        vU_lands_on = next(i for i in _GIT_ANNEX_VERSIONS_INFO["supported"] if i >= Uversion)
-
+        v_first_supported = next(i for i in _GIT_ANNEX_VERSIONS_INFO["supported"] if i >= Uversion)
         annex = AnnexRepo(path1, create=True, version=Uversion)
         assert_repo_status(path1, annex=True)
-        version = int(annex.config.get('annex.version'))
+        v_upgraded_to = int(annex.config.get('annex.version'))
 
-        eq_(version, vU_lands_on)
-        assert_in("will be upgraded to 8", cm.out)
+        if external_versions['cmd:annex'] <= '10.20220724':
+            eq_(v_upgraded_to, v_first_supported)
+            assert_in("will be upgraded to 8", cm.out)
+        else:
+            # 10.20220724-5-g63cef2ae0 started to auto-upgrade to 10, although 8 was the
+            # lowest supported. In general we can only assert that we upgrade into one
+            # of the supported
+            assert_in(v_upgraded_to, _GIT_ANNEX_VERSIONS_INFO["supported"])
+            assert_in("will be upgraded to %s or later version" % v_first_supported, cm.out)
 
     # default from config item (via env var):
     with patch.dict('os.environ', {'DATALAD_REPO_VERSION': str(Uversion)}):
+        # and check consistency of upgrading to the default version:
         annex = AnnexRepo(path2, create=True)
         version = int(annex.config.get('annex.version'))
-        eq_(version, vU_lands_on)
+        eq_(version, v_upgraded_to)
 
 
 @pytest.mark.parametrize("version", _GIT_ANNEX_VERSIONS_INFO["supported"])

